app/controllers/api/base_controller.rb

```diff
@@ -29,6 +29,11 @@ def verify_with_otp
     render plain: prompt_text, status: :unauthorized
   end
 
+  def verify_mfa_requirement
+    return if (@rubygem.mfa_required? && @api_user.mfa_enabled?) || !@rubygem.mfa_required?
```

Comment:
`return unless @rubygem.mfa_required? && !@api_user.mfa_enabled?` would be logically equivalent and would read a little better.

Reply:
I also noticed you use this in a few places - you could extract it somewhere to dry it up (helper? private method in controller? idk)

Reply:
You're right, I've refactored it a bit and moved the condition to `Rubygem#mfa_requirement_satisfied_for?(user)`.